File: data-structure-algo/Arrays/continuous_sum/ContinuousSum.java
public class ContinuousSum {
    public static void main(String args[]) {
        int[] myArray = {1,2,-1,3,4,10,10,-10,-1};

        System.out.println(maxSum(myArray));
    }

    public static int maxSum (int [] arr) {

        int max = arr[0];
        int current = arr[0];
        
        for (int i = 1; i < arr.length; i++) {
            current = Math.max(current + arr[i], arr[i]);

            max = Math.max(current, max);
        }

        return max;
    }
}

File: data-structure-algo/Arrays/continuous_sum/continuousSum.js

function continuousSum(array) {

    max_sum = current_sum = array[0] 
    
    for (num = 1; num < array.length; num++) {

        current_sum = Math.max(current_sum+array[num], array[num])    

        max_sum = Math.max(max_sum, current_sum)
    }

    return max_sum
}

console.log(continuousSum([1,2,-1,3,4,10,10,-10,-1]))
